scripts/prepare-unity.sh

```diff
@@ -45,11 +45,15 @@ if [ -d "$IMPROBABLE_UNITY_VERSIONED" ]; then
   fi
 
   # Note that on MacOS, Mac standalone support is installed within the Unity app package itself.
+  # Only check for Windows IL2CPP support on Windows agents, as Unity do not provide Windows-IL2CPP support for Mac.
+  # Only check for Mac IL2CPP support on Mac agents, as Unity do not provide Mac-IL2CPP support for Windows.
   if [ -d "$PLAYBACK_ENGINE_ROOT/LinuxStandaloneSupport" ] && \
      [ -d "$PLAYBACK_ENGINE_ROOT/AndroidPlayer" ] && \
      [ -d "$PLAYBACK_ENGINE_ROOT/iOSSupport" ] && \
      [ -d "$PLAYBACK_ENGINE_ROOT/WindowsStandaloneSupport" ] && \
-     ( isMacOS || [ -d "$PLAYBACK_ENGINE_ROOT/MacStandaloneSupport" ])
+     ( isMacOS || [ -d "$PLAYBACK_ENGINE_ROOT/MacStandaloneSupport" ]) && \
+     ( isMacOS || [ -d "$PLAYBACK_ENGINE_ROOT/WindowsStandaloneSupport/Variations/win64_development_il2cpp" ]) && \
+     ( isWindows || [ -d "$IMPROBABLE_UNITY_ROOT/Unity.app/Contents/PlaybackEngines/MacStandaloneSupport/Variations/macosx64_development_il2cpp" ])
```

Comment:
@gkassabli  just saw your previous comment

I see what you mean but then I think it becomes a lot less readable because atm it follows the "A -> B = ¬A || B" rule nicely